Source/Core/VideoCommon/ShaderCache.cpp

```diff
@@ -449,15 +449,15 @@ ShaderCache::CompileVertexUberShader(const UberShader::VertexShaderUid& uid) con
 std::unique_ptr<AbstractShader> ShaderCache::CompilePixelShader(const PixelShaderUid& uid) const
 {
   const ShaderCode source_code =
-      GeneratePixelShaderCode(m_api_type, m_host_config, uid.GetUidData(), {});
+      GeneratePixelShaderCode(m_api_type, m_host_config, uid.GetUidData(), "", "");
```

Comment:
I only have a vague understanding of this whole system but `CustomPixelShaderContents` seemed like a nicer interface to me. Now two empty strings need to be passed for a default state?

Reply:
Is it just the fact it is in a struct that is default initialized that you find attractive?  What do you think is "nicer"?

`CustomPixelShaderContents` is _not_ a good interface and was a terrible idea.  It contains a vector which added complexity (what do you do with two shaders?  Merging is slow as heck and how do you deal with conflicts?  I tried but it gets complicated and might not even be what the user wants).  Also dynamic memory.  The inner `CustomPixelShader` is again doing dynamic allocation of strings.

Also, it was already empty strings, just hidden from you in this part of the code.

Reply:
I just mean the passing of `CustomPixelShaderContents` seemed cleaner, not the internals.
All the "custom" stuff was in one parameter. Now each call-site needs to be changed if additional "custom stuff" is added?

Reply:
Ah ok.  That's reasonable.  I was trying to avoid adding more structs.  But I will add a new one to handle this.